source/transfer_list.rst

```diff
@@ -902,5 +902,172 @@ This manifest contains the SPMC attribute node consumed by the SPMD
 
+
+**Executable entry point information entry layout (XFERLIST_EXEC_EP_INFO)**
+
+This entry type holds one or more `entry_point_info_t` structures.
+
+`entry_point_info_t` is a TF-A-specific data structure [TF_BL31]_ used to
+represent the execution state of an image; that is, the state of general purpose
+registers, PC, and SPSR. Due to architectural differences in AArch64 and
+AArch32, there exist two representations of `entry_point_info_t`.
+
+.. entry_point_info_t:
+.. list-table:: entry_point_info_t structure layout (AArch64)
+   :widths: 2 2 2 8
+
+   * - field
+     - size (bytes)
+     - offset (bytes)
+     - description
+
+   * - h
+     - 0x8
+     - 0x0
+     - Structure header `param_header_t`.
+
+   * - pc
+     - 0x8
+     - 0x8
+     - Program counter (PC) pointing to the entry point of the executable.
+
+   * - spsr
+     - 0x4
+     - 0xc
+     - Holds the saved process state (SPSR) for the next executable.
+
+   * - args
+     - 0x40
+     - 0x18
+     - Holds the values of registers `x0` to `x7`.
+
+.. entry_point_info_t:
+.. list-table:: entry_point_info_t structure layout (AArch32)
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - h
+     - 0x8
+     - 0x0
+     - Header `param_header_t` holding information about the type, size, version and attributes of
+       this structure.
+
+   * - pc
+     - 0x4
+     - 0x8
+     - Program counter pointing to the entry point of the executable.
+
+   * - spsr
+     - 0x4
+     - 0xc
+     - Holds the saved process state for the next executable.
+
+   * - lr_svc
+     - 0x4
+     - 0x10
+     - Holds the link register (LR) value used during exception return in Secure
+       or Non-secure Supervisor mode.
+
+   * - args
+     - 0x40
+     - 0x14
+     - Holds the values of registers `r0` to `r3`.
+
+The header of `entry_point_info_t` holds information about the attributes of the
+executable. Bits 0 and 5 of this field are used to encode the security state;
+in other words, whether the image is to be executed in Secure, Non-Secure, or Realm
+mode.
+
+.. param_header_t:
+.. list-table:: Layout of header data structure `param_header_t`.
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - type
+     - 0x1
+     - 0x0
+     - Type of the represented structure i.e. executable info.
+
+   * - version
+     - 0x1
+     - 0x1
+     - Version of the `param_header_t` structure (version 2 in this
+       specification).
+
+   * - size
+     - 0x2
+     - 0x3
+     - Size of `param_header_t`.
+
+   * - attr
+     - 0x4
+     - 0x5
+     - Attributes of the represented structure.
+
+.. param_header_attr_state:
+.. list-table:: Security state encoding in `param_header_t` attributes for `entry_point_info_t`.
+
+   * - Bit 5
+     - Bit 0
+     - Security State
+
+   * - 0
+     - 0
+     - Secure
+
+   * - 0
+     - 1
+     - Non-Secure
+
+   * - 1
+     - 1
+     - Realm
```

Comment:
I think it's a mistake for us to explicitly specify entry_point_info and param_header header. We don't do it for any other entries (e.g. DT, HOB, SPMC manifest, TPM event log, ....) so why start now? This stuff is already specified in the external reference: https://trustedfirmware-a.readthedocs.io/en/latest/design/firmware-design.html#data-structures-used-in-the-bl31-cold-boot-interface

Admittedly, that doc could do with more detail but I think the place for that detail is there, not here. Note that entry-point_info is itself a versioned and self-describing structure so can change in future without affecting compatibility in this spec.

Reply:
Let's remove the these descriptions if everyone is in agreement. Longer term, it will create a maintenance burden on this project if the information is duplicated here. I've raised a separate patch to capture the structure descriptions in TF-A's documentation: https://review.trustedfirmware.org/c/TF-A/trusted-firmware-a/+/26233.

Reply:
Agree, lets remove the description from here.